### ISSUE TITLE
Layer issue on xaxis tooltip

### ISSUE THREAD

**Issue by jimfilippou** [apexcharts/apexcharts.js#145]
This happens when you have a lot of datasets, any thoughts on solving this?

![image](https://user-images.githubusercontent.com/6877046/46475051-1cd99c00-c7ed-11e8-95ed-41fbc987c392.png)

**Comment by jimfilippou** [comment]
Increasing the height just in case it solves it

**Comment by junedchhipa** [comment]
I think a conditional check is required. If the tooltip height exceeds grid area, the tooltip needs to stick inside the grid area which will prevent getting `x-axis tooltip` in the way of the general tooltip.

**Comment by jimfilippou** [comment]
This is the minimum that the chart y can get. The y here is 0 however the data can't fit in there. Instead of calculating Y, the tooltip's items must divide in two lists and not one long list that can't fit in the UI. So i suggest that in case of multiple datasets, the default behavior should group the datasets decreasing the height.

![untitled](https://user-images.githubusercontent.com/6877046/46484659-db53eb80-c802-11e8-8c08-44381477f941.png)

**Comment by junedchhipa** [comment]
Good suggestion, I will give it a try

**Comment by junedchhipa** [comment]
I tried making multiple columns in the tooltip - but it didn't work out great. Many things will have to be taken into consideration when changing tooltip layouts for many items.
So, for time being, I suggest using CSS (removing paddings/making font size smaller, etc) to accommodate a large number of items into the chart area.

I have prevented the tooltip to go below the chart area preventing drawing it over x-axis tooltip.
